lightningd/peer_control.c

```diff
@@ -1045,8 +1045,6 @@ static void peer_connected_hook_final(struct peer_connected_hook_payload *payloa
 		 * dualopend. we only get here if there's an error  */
 		if (channel) {
 			assert(!channel->owner);
-			assert(channel->state == DUALOPEND_OPEN_INIT
-			       || channel->state == DUALOPEND_AWAITING_LOCKIN);
```

Comment:
So I re-added this assertion to see what the problem was...

Crickets.  Hmm...